### ISSUE TITLE
`zst_offset` not warning for NonNull

### ISSUE THREAD

**Issue by esthedebeste** [rust-lang/rust-clippy#16887]
### Summary

`NonNull<T>` is a non-null guaranteeing wrapper on `*mut T`. It also has `add`, `offset`, etc. implementations. These are not checked for `NonNull<()>`.

### Lint Name

zst_offset

### Reproducer

I tried this code:

```rust
#![warn(clippy::zst_offset)]
use core::ptr::NonNull;

fn main() {
    let mut x = 3;
    let addr: NonNull<()> = NonNull::new(&mut x as *mut i32 as *mut ()).unwrap();
    let addr_plus_one = unsafe { addr.add(1) };
    println!("{addr:?} + 1 = {addr_plus_one:?}");
}
```
[playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=b5b9f2cf3e1b18623d8e4a10612e241d)

I expected to see this happen:

A warning as `addr.add(1)` is a noop (like in [this playground](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2024&gist=3519e2bf0f24269416cd1da1b2cba92d))

Instead, this happened:

No warning

### Version

```text
0.1.97 (2026-04-18 0febdbab27)
```
